InnowiseSQL.sql

```diff
@@ -1 +1,416 @@
+	   (9,300,12),
+	   (9,1031,13),
+	   (9,5000,14),
+	   (10,1166,15),
+	   (10,200,16),
+	   (10,562,17);
+
+INSERT INTO Cards(Balance,AccountId)
+VALUES (320,1),
+	   (200,1),
+	   (200,1),
+	   (200,1),
+	   (200,1),
+	   (0,4),
+	   (0,4),
+	   (0,4),
+	   (0,4),
+	   (200,24),
+	   (99,24),
+	   (531,13),
+	   (200,27),
+	   (94,27),
+	   (138,29),
+	   (100,23),
+	   (100,23),
+	   (100,23),
+	   (62,17),
+	   (0,10),
+	   (0,10),
+	   (0,14),
+	   (0,14),
+	   (0,7),
+	   (0,7),
+	   (0,7),
+	   (1,7),
+	   (1,7),
+	   (0,3),
+	   (40,25),
+	   (40,25),
+	   (40,25),
+	   (40,25),
+	   (40,25),
+	   (66,15),
+	   (100,15),
+	   (0,28),
+	   (0,9),
+	   (0,9),
+	   (0,16),
+	   (75,20),
+	   (122,22),
+	   (128,19),
+	   (100,19),
+	   (100,19),
+	   (100,19),
+	   (100,19);
+
+
+	--Query 1
+
+SELECT DISTINCT Banks.Name, Towns.Name
+FROM Banks
+INNER JOIN Filials ON Banks.Id = Filials.BankId
+INNER JOIN Buildings ON Filials.Id = Buildings.FilialId
+INNER JOIN Towns ON Buildings.TownId = Towns.Id
+WHERE Towns.Name = 'Минск'
+
+
+	--Query 2
+
+SELECT Banks.Name, Cards.Id, Accounts.Id, Clients.Name, Cards.Balance AS Card_Balance, Accounts.Balance AS Account_Balance
+FROM Banks
+INNER JOIN Accounts ON Accounts.BankId = Banks.Id 
+INNER JOIN Clients ON Accounts.ClientId = Clients.Id
+INNER JOIN Cards ON Cards.AccountId = Accounts.Id;
+
+
+	--Query 3
+
+SELECT Accounts.Id, Accounts.Balance, SUM(Cards.Balance) AS Summa, Accounts.Balance - SUM(Cards.Balance) as dif
+FROM Accounts INNER JOIN Cards ON AccountId = Accounts.Id
+GROUP BY Accounts.Id, Accounts.Balance
+HAVING SUM(Cards.Balance) != Accounts.Balance
+
+	--Query 4a
+
+SELECT SocStatus.Name, COUNT(*) AS CardsAmount
+FROM Clients
+INNER JOIN Accounts ON ClientID = Clients.Id
+INNER JOIN SocStatus ON SocStatus.Id = SocStatusID
+INNER JOIN Cards ON AccountID = Accounts.Id
+GROUP BY SocStatus.Name;
+
+	--Query 4b 
+
+
+	--Query 5
+
+CREATE PROCEDURE AccrualForSocStatus
+@status NVARCHAR
+AS
+BEGIN
+UPDATE Accounts
+SET Balance = Balance + 10
+FROM Accounts
+	INNER JOIN Clients ON Clients.Id = ClientID
+	INNER JOIN SocStatus ON SocStatusId = SocStatus.Id
+WHERE SocStatus.Name = @status
+END
+GO
+
+
+	--Query 6
+
+SELECT AccountID, Accounts.Balance, SUM(Cards.Balance) AS Summa,
+(Accounts.Balance - SUM(Cards.Balance))/COUNT(Cards.Balance) AS AvailableFunds
+FROM Accounts INNER JOIN Cards on Accounts.Id = AccountID
+GROUP BY AccountID, Accounts.Balance
+HAVING Accounts.Balance - SUM(Cards.Balance) > 0
+
+	--Query 7
+
+CREATE PROCEDURE Transfer
+@amount MONEY,
+@CrdID INT
+AS
+DECLARE @sumCards MONEY;
+SET @sumCards = (SELECT SUM(Cards.Balance)
+				 FROM Accounts INNER JOIN Cards ON Accounts.Id = AccountID
+				 WHERE Accounts.Id = (SELECT AccountId
+									  FROM Cards 
+									  WHERE Cards.Id = @CrdID));
+DECLARE @AccBlnc MONEY;
+SET @AccBlnc = (SELECT Accounts.Balance
+				FROM Accounts
+				WHERE Accounts.Id = (SELECT AccountId
+									 FROM Cards 
+									 WHERE Cards.Id = @CrdID));
```

Comment:
You make here two requests for @AccBlnc and @sumCards
You can improve it and make only one, and use only one variable